smart/Conditional.md

```diff
@@ -0,0 +1,35 @@
+# Conditional balances
+
+## Article I: OPEN PROBLEMS
+
+An _Open Problem_ is a problem which has no answer _yet_. Examples of _Open Problems_ include:
+- Is P == NP ?
+- Will the sun rise tomorrow ?
+
+## Article II: CONDITIONAL TRANSACTIONS
+
+A transaction can be conditioned on the answer of an _Open Problem_.
+For example: If P == NP then Alice gives 8 Gitcoins to Bob else Bob gives 8 Gitcoins to Alice.
+
+## Article III: CONDITIONAL BALANCES
+
+As a result of a conditional transaction, the balance of a Githereum User can itself become conditional. 
+
+For example:
+
+```
+Alice:
+  - I have 16 Gitcoins if P == NP
+  - I have 32 Gitcoins if P != NP
+```
```

Comment:
We should provide a clearer syntax to allow multi conditional balances.
For example we should be able to refer to old commit hashes and execute operations on them to provide a new of describing the balance:
```
Alice:
   - I have <commit_hash> - X Gitcoins if <condition(s)>
   - I have <commit_hash> + X Gitcoins if <condition(s)>

Reply:
I think the full state should be observable by looking at a single file, otherwise its hard to check if _the math checks out_ when performing transactions !